catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ResultIterable.java

```diff
@@ -155,18 +212,22 @@ private void fetchNextResults() {
             queryCopy.setStartIndex(currentIndex);
 
             try {
-                SourceResponse response = queryFunction.query(queryRequestCopy);
+                response = queryFunction.query(queryRequestCopy);
 
                 if (response.getResults()
                         .size() == 0) {
-                    results = RESULTS_EXHAUSTED;
+                    finished = true;
                     return;
                 }
 
                 results = response.getResults()
                         .iterator();
                 currentIndex += response.getResults()
                         .size();
+
```

Comment:
![screenshot 2017-08-15 07 02 33](https://user-images.githubusercontent.com/9013416/29319128-bbc2b6e2-8187-11e7-82f9-3ad7723c82c5.png)
I changed that yesterday, just must have missed it in the commit